hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSZKFailoverController.java

```diff
@@ -158,7 +158,7 @@ public static DFSZKFailoverController create(Configuration conf) {
           "You may run zkfc on the node other than namenode.";
       throw new HadoopIllegalArgumentException(msg);
     }
-    NameNode.initializeGenericKeys(localNNConf, nsId, nnId);
```

Comment:
In `NNHAServiceTarget` the call is on a copy of `localNNConf` though so the original conf will not be set.

Reply:
NameNode.initializeGenericKeys(), this method will be called twice. In addition to this, it will be called once when NNHAServiceTarget() is initialized. Therefore, it can be deleted here.

Reply:
Yes but as I said, the call in `NNHAServiceTarget` is on a copy:
```java
    // Make a copy of the conf, and override configs based on the
    // target node -- not the node we happen to be running on.
    HdfsConfiguration targetConf = new HdfsConfiguration(conf);
    NameNode.initializeGenericKeys(targetConf, nsId, nnId);
```
so the original `localNNConf` won't have those set. I'm not sure there will be any side effect by doing this.

Reply:
I double checked. I approve of your ideas.
thank you very much.